presto-hive/src/main/java/com/facebook/presto/hive/metastore/PrestoTableType.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.metastore;
+
+import java.util.Optional;
+
+public enum PrestoTableType
+{
+    MANAGED_TABLE,
+    EXTERNAL_TABLE,
+    VIRTUAL_VIEW,
+    OTHER,
+    /**/;
+
+    public static Optional<PrestoTableType> optionalValueOf(String value)
+    {
+        for (PrestoTableType type : PrestoTableType.values()) {
```

Comment:
We can use a Map for lookup? i.e. 

```java
    private static final Map<String, PrestoTableType> TABLE_TYPE_BY_NAME = ImmutableMap.of(
            MANAGED_TABLE.name(), MANAGED_TABLE,
            EXTERNAL_TABLE.name(), EXTERNAL_TABLE,
            VIRTUAL_VIEW.name(), VIRTUAL_VIEW,
            TEMPORARY_TABLE.name(), TEMPORARY_TABLE);

    public static PrestoTableType getTableTypeByName(String value)
    {
        return TABLE_TYPE_BY_NAME.getOrDefault(value.toUpperCase(), OTHER);
    }
```

Reply:
> Map<String, PrestoTableType> TABLE_TYPE_BY_NAME

I theory a simple loop is more efficient in this context.

HashMap lookup requires taking a hash from the input value, that is on average ~10 symbols long. 

Given that the first letter is different for common values, for the common case looping will require at most 4 single symbol comparisons to find the value. (Of course the full match has to be checked at the end of the day, but it has to be done in the hash map lookup as well)

Given that the loop simply requires less code, i would like to stick with it.

> value.toUpperCase()

I was trying to mimic the existing behaviour. Currently the match is case sensitive. 

See:

https://github.com/prestodb/presto/pull/12464/commits/a1a1bffffe7140ef46d7cae5fafc9fe510255a1e#diff-13aa0bdc2b255255b85a1b0763c2cb6dL444
https://github.com/prestodb/presto/pull/12464/commits/a1a1bffffe7140ef46d7cae5fafc9fe510255a1e#diff-404dab2bb7ab457656fcd803f292480eL88